chartfx-chart/src/main/java/de/gsi/chart/renderer/spi/financial/CandleStickRenderer.java

```diff
@@ -0,0 +1,317 @@
+public class CandleStickRenderer extends AbstractFinancialRenderer<CandleStickRenderer> implements Renderer {
+    private final boolean paintVolume;
+    private final FindAreaDistances findAreaDistances;
+
+    protected List<RendererPaintAfterEP> paintAfterEPS = new ArrayList<>();
+
+    public CandleStickRenderer(boolean paintVolume) {
+        this.paintVolume = paintVolume;
+        this.findAreaDistances = paintVolume ? new XMinVolumeMaxAreaDistances() : new XMinAreaDistances();
+    }
+
+    public CandleStickRenderer() {
+        this(false);
+    }
+
+    public boolean isPaintVolume() {
+        return paintVolume;
+    }
+
+    @Override
+    public Canvas drawLegendSymbol(DataSet dataSet, int dsIndex, int width, int height) {
+        final Canvas canvas = new Canvas(width, height);
+        final GraphicsContext gc = canvas.getGraphicsContext2D();
+        final String style = dataSet.getStyle();
+
+        gc.save();
+        Color candleLongColor = StyleParser.getColorPropertyValue(style, DATASET_CANDLESTICK_LONG_COLOR, Color.GREEN);
+        Color candleShortColor = StyleParser.getColorPropertyValue(style, DATASET_CANDLESTICK_SHORT_COLOR, Color.RED);
+
+        gc.setFill(candleLongColor);
+        gc.setStroke(candleLongColor);
+        gc.fillRect(1, 3, width / 2.0 - 2.0, height - 8.0);
+        double x = width / 4.0;
+        gc.strokeLine(x, 1, x, height - 2.0);
+
+        gc.setFill(candleShortColor);
+        gc.setStroke(candleShortColor);
+        gc.fillRect(width / 2.0 + 2.0, 4, width - 2.0, height - 12.0);
+        x = 3.0 * width / 4.0 + 1.5;
+        gc.strokeLine(x, 1, x, height - 3.0);
+        gc.restore();
+
+        return canvas;
+    }
+
+    @Override
+    protected CandleStickRenderer getThis() {
+        return this;
+    }
+
+    @Override
+    public List<DataSet> render(final GraphicsContext gc, final Chart chart, final int dataSetOffset,
+            final ObservableList<DataSet> datasets) {
+        if (!(chart instanceof XYChart)) {
+            throw new InvalidParameterException(
+                    "must be derivative of XYChart for renderer - " + this.getClass().getSimpleName());
+        }
+        final XYChart xyChart = (XYChart) chart;
+
+        // make local copy and add renderer specific data sets
+        final List<DataSet> localDataSetList = new ArrayList<>(datasets);
+        localDataSetList.addAll(super.getDatasets());
+
+        long start = 0;
+        if (ProcessingProfiler.getDebugState()) {
+            start = ProcessingProfiler.getTimeStamp();
+        }
+
+        final Axis xAxis = xyChart.getXAxis();
+        final Axis yAxis = xyChart.getYAxis();
+
+        final double xAxisWidth = xAxis.getWidth();
+        final double xmin = xAxis.getValueForDisplay(0);
+        final double xmax = xAxis.getValueForDisplay(xAxisWidth);
+        int index = 0;
+
+        for (final DataSet ds : localDataSetList) {
+            if (ds.getDimension() > 7)
```

Comment:
```suggestion
            if (ds.getDimension() < 7) // dataset has to have at least 7 dimensions, time -> (open, high, low, close, volume, oi)
```

Additional dimensions should be ignored, but if you have less than 7 you will get index out of bounds errors later on. As the renderer only seems to use the first 4  y values, `< 5 `could also be a valid choice.

Probably the same for HiLowRenderer?